torch/csrc/autograd/variable.h

```diff
@@ -592,12 +552,13 @@ inline Variable make_variable_consuming(
     bool allow_tensor_metadata_change = true) {
   TORCH_CHECK(
       !data.is_variable(),
-      "Must not create a new variable from a variable, use its .data()");
+      "Must not create a new variable from a variable, use its .tensor_data()");
   if (data.defined()) {
     AT_ASSERT(data.getIntrusivePtr().use_count() == 1);
-    data.unsafeGetTensorImpl()->set_allow_tensor_metadata_change(allow_tensor_metadata_change);
-    auto autograd_meta = c10::guts::make_unique<Variable::AutogradMeta>();
-    return Variable(c10::make_intrusive<Variable::Impl>(std::move(data), std::move(autograd_meta), requires_grad));
+    auto data_impl = data.getIntrusivePtr();
```

Comment:
why does this call `getIntrusivePtr` instead of `unsafeGetTensorImpl`?

Reply:
Fixed.

Reply:
This calls `auto data_impl = data.getIntrusivePtr()` because later we need to call `Variable(std::move(data_impl))` to create the Variable, and the Variable constructor only accepts intrusive_ptr: `Variable(c10::intrusive_ptr<at::TensorImpl> self)` (which internally uses the Tensor constructor that only accepts intrusive_ptr https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/core/Tensor.h#L59-L65).